lib/ansible/module_utils/docker/swarm.py

```diff
@@ -166,6 +167,17 @@ def get_node_inspect(self, node_id=None, skip_missing=False):
 
         json_str = json.dumps(node_info, ensure_ascii=False)
         node_info = json.loads(json_str)
+
+        if 'ManagerStatus' in node_info:
+            if node_info['ManagerStatus'].get('Leader'):
+                # This is workaround of bug in Docker when in some cases the Leader IP is 0.0.0.0
+                # Check moby/moby#35437 for details
+                count_colons = node_info['ManagerStatus']['Addr'].count(":")
+                if count_colons == 1:
+                    swarm_leader_ip = split(":", node_info['ManagerStatus']['Addr'])[0] or node_info['Status']['Addr']
```

Comment:
Please do not use `re.split()` in such simple cases. Use `str.split()` instead. That's way more efficient.
(Also, use `maxsplit`: `node_info['ManagerStatus']['Addr'].split(':', 1)[0]`)